analyzer/runtime/extract.go

```diff
@@ -1037,6 +1056,209 @@ func extractEvents(blockData *BlockData, relatedAccountAddresses map[apiTypes.Ad
 					}
 					return nil
 				},
+				IUniswapV2FactoryPairCreated: func(token0ECAddr ethCommon.Address, token1ECAddr ethCommon.Address, pairECAddr ethCommon.Address, allPairsLength *big.Int) error {
+					token0Addr, err := addresses.RegisterRelatedEthAddress(blockData.AddressPreimages, relatedAccountAddresses, token0ECAddr.Bytes())
+					if err != nil {
+						return fmt.Errorf("token0: %w", err)
+					}
+					eventData.RelatedAddresses[token0Addr] = struct{}{}
+					token1Addr, err := addresses.RegisterRelatedEthAddress(blockData.AddressPreimages, relatedAccountAddresses, token1ECAddr.Bytes())
+					if err != nil {
+						return fmt.Errorf("token1: %w", err)
+					}
+					eventData.RelatedAddresses[token1Addr] = struct{}{}
+					pairAddr, err := addresses.RegisterRelatedEthAddress(blockData.AddressPreimages, relatedAccountAddresses, pairECAddr.Bytes())
+					if err != nil {
+						return fmt.Errorf("pair: %w", err)
+					}
+					eventData.RelatedAddresses[pairAddr] = struct{}{}
+					blockData.SwapCreations[SwapCreationKey{
+						Factory: eventAddr,
+						Token0:  token0Addr,
+						Token1:  token1Addr,
```

Comment:
Does uniswap enforce some ordering on token0/token1? Like is it possible to create a usdt/wrose pair and then create a wrose/usdt pair that's distinct from it? (and if uniswap combines the pairs, would nexus do the same?)

Reply:
my reading of the actual contract is they have to have the lower token address in token0